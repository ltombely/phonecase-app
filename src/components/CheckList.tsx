import { Check } from "lucide-react";
import { useTranslations } from "next-intl";

export default function CheckList() {
  const t = useTranslations();
  return (
    <ul className="mt-8 space-y-2 text-left font-medium flex flex-col items-center sm:items-start">
      <div className="space-y-2">
        <li className="flex gap-1.5 items-center text-left">
          <Check className="h-5 w-5 shrink-0 text-green-600" />
          {t("High Quality, durable material")}
        </li>
        <li className="flex gap-1.5 items-center text-left">
          <Check className="h-5 w-5 shrink-0 text-green-600" />
          {t("5 year print guarantee")}
        </li>
        <li className="flex gap-1.5 items-center text-left">
          <Check className="h-5 w-5 shrink-0 text-green-600" />
          {t("Modern iPhone models supported")}
        </li>
      </div>
    </ul>
  );
}
