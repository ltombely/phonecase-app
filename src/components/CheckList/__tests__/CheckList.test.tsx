import { render, screen } from "@testing-library/react";
import CheckList from "../CheckList";

describe("CheckList component", () => {
    it("should render the component correctly", () => {
        const tree = render(<CheckList />);
        expect(tree).toMatchSnapshot();
    });

    it("should render all 3 checks", () => {
        render(<CheckList />);

        const listItems = screen.getAllByRole("listitem");

        expect(listItems).toHaveLength(3);
    });
});
