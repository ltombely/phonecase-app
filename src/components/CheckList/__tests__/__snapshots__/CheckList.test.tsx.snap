// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckList component should render the component correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="mt-8 space-y-2 text-left font-medium flex flex-col items-center sm:items-start"
      >
        <div
          class="space-y-2"
        >
          <li
            class="flex gap-1.5 items-center text-left"
          >
            <svg
              class="lucide lucide-check h-5 w-5 shrink-0 text-green-600"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 6 9 17l-5-5"
              />
            </svg>
            High Quality, durable material
          </li>
          <li
            class="flex gap-1.5 items-center text-left"
          >
            <svg
              class="lucide lucide-check h-5 w-5 shrink-0 text-green-600"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 6 9 17l-5-5"
              />
            </svg>
            5 year print guarantee
          </li>
          <li
            class="flex gap-1.5 items-center text-left"
          >
            <svg
              class="lucide lucide-check h-5 w-5 shrink-0 text-green-600"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 6 9 17l-5-5"
              />
            </svg>
            Modern iPhone models supported
          </li>
        </div>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="mt-8 space-y-2 text-left font-medium flex flex-col items-center sm:items-start"
    >
      <div
        class="space-y-2"
      >
        <li
          class="flex gap-1.5 items-center text-left"
        >
          <svg
            class="lucide lucide-check h-5 w-5 shrink-0 text-green-600"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 6 9 17l-5-5"
            />
          </svg>
          High Quality, durable material
        </li>
        <li
          class="flex gap-1.5 items-center text-left"
        >
          <svg
            class="lucide lucide-check h-5 w-5 shrink-0 text-green-600"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 6 9 17l-5-5"
            />
          </svg>
          5 year print guarantee
        </li>
        <li
          class="flex gap-1.5 items-center text-left"
        >
          <svg
            class="lucide lucide-check h-5 w-5 shrink-0 text-green-600"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 6 9 17l-5-5"
            />
          </svg>
          Modern iPhone models supported
        </li>
      </div>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
